kge/model/kge_model.py

```diff
@@ -591,19 +591,44 @@ def penalty(self, **kwargs) -> List[Tensor]:
         # weighted).
         if "batch" in kwargs and "triples" in kwargs["batch"]:
             triples = kwargs["batch"]["triples"].to(self.config.get("job.device"))
-            return (
-                super().penalty(**kwargs)
-                + self.get_s_embedder().penalty(indexes=triples[:, S], **kwargs)
-                + self.get_p_embedder().penalty(indexes=triples[:, P], **kwargs)
-                + self.get_o_embedder().penalty(indexes=triples[:, O], **kwargs)
+            penalty_result = super().penalty(**kwargs) + self.get_p_embedder().penalty(
+                indexes=triples[:, P], **kwargs
             )
+            if self.get_s_embedder() == self.get_o_embedder():
```

Comment:
This should be "is" instead of "==": we check for object identity instead of comparing everything.